contract_show_recurring_invoice/__init__.py

```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+from . import models
+
+
+def copy_recurring_invoice(cr, registry):
+    """Copy recurring invoice on contract."""
+    cr.execute("UPDATE account_invoice "
+               "SET contract_id=aaa.id "
+               "FROM account_analytic_account aaa "
+               "WHERE aaa.code = account_invoice.origin")
```

Comment:
This sql is executed in post install, so all invoices with origin equals account analytic code must be updated, because contract_id is a new field.